sktime/forecasting/conditional_invertible_neural_network.py

```diff
@@ -217,12 +229,20 @@ def _fit(self, y, fh, X=None):
             truncate="fill",
         ).fit_transform(y)
 
-        self.function = CurveFitForecaster(
-            self._f_statistic,
-            {"p0": self._init_param_f_statistic},
-            normalise_index=True,
-        )
-        self.function.fit(rolling_mean.dropna())
+        # Curve fit with error handling
+        try:
+            self.function = CurveFitForecaster(
+                self._f_statistic,
+                {"p0": self._init_param_f_statistic},
+                normalise_index=True,
+            )
+            self.function.fit(rolling_mean.dropna())
```

Comment:
You need only to wrap the fit function or?

Reply:
this part code is already present in current sktime code repository 
![image](https://github.com/user-attachments/assets/6fb5eca6-c977-4fa6-aca6-5dbd95fe5b54)

Reply:
Sorry I was unclear. I mean wrap by the try block. I.e. only the fit call needs to be in the try block.

Reply:
done !